aspnetcore/host-and-deploy/docker/visual-studio-tools-for-docker.md

```diff
@@ -46,53 +48,119 @@ If the target framework is .NET Core, the **OS** drop-down allows for the select
+### Docker Compose
 
-The *docker-compose.yml* file contains the name of the image that's created when the project runs:
+The Visual Studio Tools for Docker add a *docker-compose* project to the solution with the following files:
 
-[!code-yaml[](visual-studio-tools-for-docker/samples/HelloDockerTools/docker-compose.yml?highlight=5)]
+* *docker-compose.dcproj* &mdash; The file representing the project. Includes a `<DockerTargetOS>` element specifying the OS to be used.
+* *.dockerignore* &mdash; Lists the file and directory patterns to exclude when generating a build context.
+* *docker-compose.yml* &mdash; The base [Docker Compose](https://docs.docker.com/compose/overview/) file used to define the collection of images to be built and run with `docker-compose build` and `docker-compose run`, respectively.
+* *docker-compose.override.yml* &mdash; An optional file, read by Docker Compose, with configuration overrides for services. Visual Studio executes `docker-compose -f "docker-compose.yml" -f "docker-compose.override.yml"` to merge these files.
+
+The *docker-compose.yml* file references the name of the image that's created when the project runs:
+
+[!code-yaml[](visual-studio-tools-for-docker/samples/2.0/docker-compose.yml?highlight=5)]
 
 In the preceding example, `image: hellodockertools` generates the image `hellodockertools:dev` when the app runs in **Debug** mode. The `hellodockertools:latest` image is generated when the app runs in **Release** mode.
 
-Prefix the image name with the [Docker Hub](https://hub.docker.com/) username (for example, `dockerhubusername/hellodockertools`) if the image will be pushed to the registry. Alternatively, change the image name to include the private registry URL (for example, `privateregistry.domain.com/hellodockertools`) depending on the configuration.
+Prefix the image name with the [Docker Hub](https://hub.docker.com/) username (for example, `dockerhubusername/hellodockertools`) if the image is to be pushed to the registry. Alternatively, change the image name to include the private registry URL (for example, `privateregistry.domain.com/hellodockertools`) depending on the configuration.
+
+### Service Fabric
+
+In addition to the base [Prerequisites](#prerequisites), the [Service Fabric](/azure/service-fabric/) orchestration solution demands the following prerequisites:
+
+* [Microsoft Azure Service Fabric SDK](https://www.microsoft.com/web/handlers/webpi.ashx?command=getinstallerredirect&appid=MicrosoftAzure-ServiceFabric-CoreSDK) version 2.6 or later
+* Visual Studio 2017's **Azure Development** workload
+
+Service Fabric doesn't support running Linux containers in the local development cluster on Windows. If the project is already using a Linux container, Visual Studio prompts to switch to Windows containers.
+
+The Visual Studio Tools for Docker do the following tasks:
```

Comment:
Maybe "perform" over the more informal "do"

Reply:
I preferred perform too, but Acrolinx didn't.

Reply:
*R2-D2 you know better than to trust a strange computer.* -- C-3PO (Star Wars Episode V: The Empire Strikes Back)